x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/MulTests.java

```diff
@@ -124,7 +125,20 @@ public static Iterable<Object[]> parameters() {
             )
         );
 
-        return parameterSuppliersFromTypedData(anyNullIsNull(false, suppliers));
+        suppliers = errorsForCasesWithoutExamples(anyNullIsNull(true, suppliers), MulTests::mulErrorMessageString);
+
+        // Cannot use parameterSuppliersFromTypedDataWithDefaultChecks as error messages are non-trivial
+        return parameterSuppliersFromTypedData(suppliers);
+    }
+
+    private static String mulErrorMessageString(boolean includeOrdinal, List<Set<DataType>> validPerPosition, List<DataType> types) {
```

Comment:
Are all of the arithmetic things like this? They all have that common casting logic so maybe so?

Reply:
Most of them are similar in that they require the same type in both fields _for most cases_ (Datetime fields make this a bit more complex).
Other like Sub have more weird logic. I tried to make them all "look similar" at least.

Reply:
I think this is an improvement.  The error message logic for the arithmetic operators is unfortunately very complex. I think our testing abstractions for error messages are still not right (not something this PR is responsible for, to be clear), in that we have both `TypeErrorMessageSupplier` and `PositionalErrorMessageSupplier` which are similar and related but not the same.  I feel like we could probably merge those into one abstraction that works for all (most?) functions, but it also sounds like a lot of work for dubious gains.  Maybe a spacetime project.